tests/analyze_assoc.R

```diff
@@ -0,0 +1,128 @@
+    make_option(c("--params", "-p"), default = "params.csv",
+                help = "Filename for RAPPOR parameters")
+  )
+  opts <- parse_args(OptionParser(option_list = option_list))
+}    
+
+source("../analysis/R/encode.R")
+source("../analysis/R/decode.R")
+source("../analysis/R/simulation.R")
+source("../analysis/R/read_input.R")
+source("../analysis/R/association.R")
+
+# This function processes the maps loaded using ReadMapFile
+# Association analysis requires a map object with a map
+# field that has the map split into cohorts and an rmap field
+# that has all the cohorts combined
+# Arguments: 
+#       map = map object with cohorts as sparse matrix in
+#             object map$map
+#             This is the expected object from ReadMapFile
+#       params = data field with parameters
+# TODO(pseudorandom): move this functionality to ReadMapFile
+ProcessMap <- function(map, params) {
+  map$rmap <- map$map
+  split_map <- function(i, map_struct) {
+    numbits <- params$k
+    indices <- which(as.matrix(
+      map_struct[((i - 1) * numbits + 1):(i * numbits),]) == TRUE,
+      arr.ind = TRUE)
+    sparseMatrix(indices[, "row"], indices[, "col"],
+                 dims = c(numbits, max(indices[, "col"])))
+  }
+  map$map <- lapply(1:params$m, function(i) split_map(i, map$rmap))
+  map
+}
+
+main <- function(opts) {
+  ptm <- proc.time()
+  
+  params <- ReadParameterFile(opts$params)
+  opts_map <- list()
+  opts_map[[1]] <- opts$map1
+  opts_map[[2]] <- opts$map2
+  map <- lapply(1:2, function(x) 
+                  ProcessMap(ReadMapFile(opts_map[[x]], params = params),
+                             params = params))
+  # Reports must be of the format
+  #     cohort no, rappor bitstring 1, rappor bitstring 2
+  reportsObj <- read.csv(opts$reports, 
+                         colClasses = c("integer", "character", "character"),
+                         header = FALSE)
+  
+  # Parsing reportsObj
+  # ComputeDistributionEM allows for different sets of cohorts
+  # for each variable. Here, both sets of cohorts are identical
+  cohorts <- list()
+  cohorts[[1]] <- as.list(reportsObj[1])[[1]]
+  cohorts[[2]] <- cohorts[[1]]
+  # Parse reports from reportObj cols 2 and 3
+  reports <- lapply(1:2, function(x) as.list(reportsObj[x + 1]))
+  
+  # Split strings into bit arrays (as required by assoc analysis)
+  reports <- lapply(1:2, function(i) {
+    # apply the following function to each of reports[[1]] and reports[[2]]
+    lapply(reports[[i]][[1]], function(x) {
+      # function splits strings and converts them to numeric values  
+      as.numeric(strsplit(x, split = "")[[1]])
+    })
+  })
+  
+  joint_dist <- ComputeDistributionEM(reports, cohorts, map, 
+                                      ignore_other = TRUE,
+                                      params, marginals = NULL,
+                                      estimate_var = FALSE)
+  print("JOINT_DIST$FIT")
+  print(joint_dist$fit)
```

Comment:
I think this should go to a file like analyze.R.  Because R prints all sorts of other stuff on stderr/stdout, so it won't be easy to extract from a script.